facia/public/facebook_ia_remote.html

```diff
@@ -0,0 +1,172 @@
+                var ref = slot.attributes;
+                for (i = 0, len = ref.length; i < len; i++) {
+                    attr = ref[i];
+                    match = attr.name.match(/^data-ad-target-(.+)$/);
+                    if (match != null) {
+                        key = match[1];
+                        value = attr.value.replace(/\s*,\s*/g, "','");
+                        targeting += ".setTargeting('" + key + "',['" + value + "'])";
+                    }
+                }
+                return targeting;
+            }
+
+            function buildSegments() {
+                var filling, kruxSegments, output, ref;
+                var targeting = "";
+                kruxSegments = (ref = localStorage.getItem('kxsegs')) != null ? ref : "";
+                if (kruxSegments !== "") {
+                    filling = kruxSegments.split(',').join("','");
+                    output = "['" + filling + "']";
+                    targeting += ".setTargeting('x', " + output + ")";
+                }
+                return targeting;
+            }
+
+            var pageLevelCodePart1 = "var googletag = googletag || {}; googletag.cmd = googletag.cmd || []; (function() { var gads = document.createElement(\"script\"); gads.async = true; gads.type = \"text/javascript\"; var useSSL = \"https:\" == document.location.protocol; gads.src = (useSSL ? \"https:\" : \"http:\") + \"//www.googletagservices.com/tag/js/gpt.js\"; var node = document.getElementsByTagName(\"script\")[0]; node.parentNode.insertBefore(gads, node); })();";
+
+            var pageLevelCodePart2 = "googletag.cmd.push(function() { googletag.pubads().enableAsyncRendering(); googletag.pubads().collapseEmptyDivs(); googletag.enableServices(); });";
+
+            function buildSlotDeclaration(size) {
+                var adUnitName = slot.getAttribute("data-ad-unit");
+                return "googletag.cmd.push(function() { googletag.defineSlot('/" + networkId + "/" + adUnitName + "', " + (buildTargetSize(size)) + ", '" + slotName + "')" + (buildCustomTargeting()) + (buildSegments()) + ".addService(googletag.pubads()); });";
+            }
+
+            function buildSlotCode() {
+                return "googletag.cmd.push(function() { googletag.display('" + slotName + "'); });";
+            }
+
+            function insertScript(content) {
+                var script = document.createElement("script");
+                slot.parentElement.appendChild(script);
+                script.type = "text/javascript";
+                return script.text = content;
+            }
+
+            function insertAdConfig() {
+                var isFirstAd = scripts.length === 1;
+                if (isFirstAd) {
+                    insertScript(pageLevelCodePart1);
+                    insertScript(pageLevelCodePart2);
+                }
+                var size = slot.getAttribute("data-ad-size");
+                return insertScript(buildSlotDeclaration(size));
```

Comment:
@kelvin-chappell out of curiosity, do all these scripts need to be separate? It would be ideal to decrease the amount of dom manipulations that we have and make only one script tag with all of these in it. What do you think?